server/cloudformation/ports.go

```diff
@@ -14,36 +14,34 @@ type portAllocator interface {
 	Put(port int64) error
 }
 
-// InstancePortsProvisioner is a Provisioner that allocates instance ports.
-type InstancePortsProvisioner struct {
+// InstancePortsResource is a Provisioner that allocates instance ports.
+type InstancePortsResource struct {
 	ports portAllocator
 }
 
-func (p *InstancePortsProvisioner) Properties() interface{} {
+func newInstancePortsProvisioner(resource *InstancePortsResource) *provisioner {
+	return &provisioner{
+		Create: resource.Create,
+		Delete: resource.Delete,
+	}
+}
+
+func (p *InstancePortsResource) Properties() interface{} {
 	return nil
 }
 
-func (p *InstancePortsProvisioner) Provision(_ context.Context, req customresources.Request) (id string, data interface{}, err error) {
-	switch req.RequestType {
-	case customresources.Create:
-		var port int64
-		port, err = p.ports.Get()
-		if err != nil {
-			return
-		}
-		id = fmt.Sprintf("%d", port)
-		data = map[string]int64{"InstancePort": port}
-	case customresources.Delete:
-		port, err2 := strconv.Atoi(req.PhysicalResourceId)
-		if err2 != nil {
-			err = fmt.Errorf("physical resource id should have been a port number: %v", err2)
-			return
-		}
-		id = req.PhysicalResourceId
-		err = p.ports.Put(int64(port))
-	default:
-		err = fmt.Errorf("%s is not supported", req.RequestType)
-	}
+func (p *InstancePortsResource) Create(_ context.Context, req customresources.Request) (string, interface{}, error) {
```

Comment:
Man - does your linter not freak out at you for not having comments describing exported methods?  It gets annoying - I'm wondering if I should just find a way to disable it, or if we should try and get better about listening to it.